### PR TITLE
Only call update if no other source prop other than data differs

### LINKED ISSUE #5731: setStyle diffing does not work if source gets clustered on the fly
**mapbox-gl-js version**: 0.42.1
### Steps to Trigger Behavior

 1. have a map with a source with cluster false
 2. set the source to cluster true on the fly after the map is initialised
 3. apply the new style on the map with setStyle

### Expected Behavior
The points should appear clustered

### Actual Behavior
The points do not appear clustered

If I start with cluster: true from the beginning all is fine. Also disabling the style diffing kind of works (ofcourse I get flashes on the map, but the clustering behaviour is as I would expect).

### PR DESCRIPTION
**Pull request by bartvde**
This PR fixes #5731

### CODE REVIEW

**Comment by bartvde** [comment]
Good point, will adapt

**Comment by bartvde** [comment]
@mourner I added https://github.com/mapbox/mapbox-gl-js/pull/5745/commits/a6f57dae933ede6422b22dcca380abd711d35b85 but I'm unsure if it's okay to use union from lodash

**Comment by bartvde** [comment]
thanks for the guidance @mourner I updated the PR

File: src/style-spec/diff.js [hunk 1]
```diff
@@ -111,25 +141,25 @@ function diffSources(before, after, commands, sourcesRemoved) {
...
+            addSource(sourceId, after, commands);
         } else if (!isEqual(before[sourceId], after[sourceId])) {
             if (before[sourceId].type === 'geojson' && after[sourceId].type === 'geojson') {
-                // geojson sources use setGeoJSONSourceData command to update
-                commands.push({ command: operations.setGeoJSONSourceData, args: [sourceId, after[sourceId].data] });
+                if (canUpdateGeoJSON(before, after, sourceId)) {
+                    commands.push({ command: operations.setGeoJSONSourceData, args: [sourceId, after[sourceId].data] });
+                } else {
+                    updateSource(sourceId, after, commands, sourcesRemoved);
+                }
```
Comment (mourner) [inline src/style-spec/diff.js]: Nit: you can move the `canUpdateGeoJSON` condition to the parent `if`, which will remove the need for the `else` above.

Comment (bartvde) [reply src/style-spec/diff.js]: good point will do

Comment (bartvde) [reply src/style-spec/diff.js]: changes made

**Pull request merged**